src/views/CardStack.js

```diff
@@ -189,39 +183,48 @@ class CardStack extends Component<DefaultProps, Props, void> {
     transitionProps: NavigationTransitionProps,
     headerMode: HeaderMode
   ): ?React.Element<*> {
-    const headerConfig = this.props.router.getScreenConfig(
+    const headerConfig: HeaderConfig = this.props.router.getScreenConfig(
       transitionProps.navigation,
       'header'
     ) || {};
 
-    return (
-      <this.props.headerComponent
-        {...transitionProps}
-        router={this.props.router}
-        style={headerConfig.style}
-        mode={headerMode}
-        onNavigateBack={() => this.props.navigation.goBack(null)}
-        renderLeftComponent={(props: NavigationTransitionProps) => {
-          const header = this.props.router.getScreenConfig(props.navigation, 'header') || {};
-          return header.left;
-        }}
-        renderRightComponent={(props: NavigationTransitionProps) => {
-          const header = this.props.router.getScreenConfig(props.navigation, 'header') || {};
-          return header.right;
-        }}
-        renderTitleComponent={(props: NavigationTransitionProps) => {
-          const header = this.props.router.getScreenConfig(props.navigation, 'header') || {};
-          // When we return 'undefined' from 'renderXComponent', header treats them as not
-          // specified and default 'renderXComponent' functions are used. In case of 'title',
-          // we return 'undefined' in case of 'string' too because the default 'renderTitle'
-          // function in header handles them.
-          if (typeof header.title === 'string') {
-            return undefined;
-          }
-          return header.title;
-        }}
-      />
-    );
+    if (
+      typeof headerConfig.bar !== 'undefined'
+      && typeof headerConfig.bar !== 'function'
+    ) {
+      return headerConfig.bar;
+    }
+
+    const renderHeader = headerConfig.bar || ((props: *) => <Header {...props} />);
```

Comment:
@ericvicenti we don't support components to align with the rest of the API (see left/right for reference). `React.Element` is accepted (by default), see lines 192-196. Function is accepted when `bar` has to be set based on props (generic implementation or any other use-case).